src/NUnitTestAdapter/TestConverter.cs

```diff
@@ -159,14 +159,24 @@ public DiaNavigationData GetNavigationData(string className, string methodName)
         {
             if (this.DiaSession == null) return null;
 
+            // First try using the class and method names provided directly
             var navData = DiaSession.GetNavigationData(className, methodName);
```

Comment:
This is true so long as you are not creating the AppDomain unnecessarily and it's already created if NavigationDataHelper is non null. I thought about doing what you suggest but i was afraid the code would be hard to read. Let me see if I can change the names of some variables and properties so it is not confusing.